docs/CONTRIBUTING_DEVELOPMENT.md

```diff
@@ -0,0 +1,262 @@
+# Contribution process for developers
+
+If you plan on contributing code to Clarity, please follow this step-by-step
+process to reduce the risk of significant changes being requested when you
+submit your pull request. We'll work with you on every step and try to be as
+responsive as possible.
+
+## I. Proposal template
+
+Before you start coding anything, please fill out the following proposal
+template in as much detail as you can. The more complete your details, the
+better, but not all questions will apply for every change. As you fill this out,
+please make sure to follow our coding guidelines for components.
+
+* [Angular Guidelines](/docs/CODING_GUIDELINES_ANGULAR.md)
+* [Core Web Component Guidelines](/docs/CODING_GUIDELINES_CORE.md)
+
+```markdown
+## Summary
+
+Describe the change or feature in detail. Try to answer the following questions.
+
+* What is the change?
+* What is a use case for this change?
+* Why should it go in Clarity?
+* Is this a change to an Angular Component or a Web Component?
+* Does this change impact existing behaviors? If so, how?
+* If this change introduces a new behavior, is this behavior accessible?
+
+## Examples
+
+_If possible, show examples of the change. You may create one yourself, or link to external sites that have the idea. It can also be beneficial to prototype the idea in isolation outside of Clarity with a Plunkr or Stackblitz example._
+
+## API
+
+_Describe the intended API for the feature you want to add. This would include:_
+
+* CSS classes and DOM structure for pure static UI contribution
+* If Angular any inputs/outputs, components, directives, services, or anything that is exported publicly for Angular contributions.
```

Comment:
`If Angular: any inputs/outputs, components, directives, services, or anything that is exported publicly for Angular contributions.`